prov/gni/src/gnix_vc.c

```diff
@@ -1577,22 +1582,12 @@ static struct gnix_vc *__gnix_nic_next_pending_rx_vc(struct gnix_nic *nic)
 /* Progress VC RXs.  Exit when all VCs are empty or if an error is encountered
  * during progress.  Failure to process an RX on any VC likely indicates an
  * inability to progress other VCs (due to low memory). */
-static int __gnix_nic_vc_rx_progress(struct gnix_nic *nic)
+static int __gnix_vc_nic_rx_progress(struct gnix_nic *nic)
 {
-	int ret;
-	struct gnix_vc *first_vc = NULL, *vc;
+	struct gnix_vc *vc;
 
 	while ((vc = __gnix_nic_next_pending_rx_vc(nic))) {
-		ret = __gnix_vc_rx_progress(vc);
-		if (ret != FI_SUCCESS)
-			break;
-
-		if (!first_vc) {
-			/* Record first VC processed. */
-			first_vc = vc;
-		} else if (vc == first_vc) {
-			/* VCs can self reschedule or be rescheduled in other
-			 * threads.  Exit if we loop back to the first VC. */
+		if (__gnix_vc_rx_progress(vc) != FI_SUCCESS) {
```

Comment:
It's not the direct goal, but we need some way to break out of the loop without keeping track of every single VC that's been scheduled during a trip thru the progress loop.